tests/core/contracts/test_contract_ambiguous_events.py

```diff
@@ -0,0 +1,260 @@
+            "type": "address",
+        },
+        {
+            "indexed": True,
+            "internalType": "address",
+            "name": "to",
+            "type": "address",
+        },
+        {
+            "indexed": False,
+            "internalType": "uint256",
+            "name": "value",
+            "type": "uint256",
+        },
+    ],
+    "name": "Transfer",
+    "type": "event",
+}
+ABI_EVENT_DEPOSIT_WITH_TUPLE = {
+    "anonymous": False,
+    "inputs": [
+        {
+            "indexed": True,
+            "internalType": "address",
+            "name": "from",
+            "type": "address",
+        },
+        {
+            "indexed": True,
+            "internalType": "bytes32",
+            "name": "id",
+            "type": "bytes32",
+        },
+        {
+            "components": [
+                {"internalType": "uint256", "name": "x", "type": "uint256"},
+                {"internalType": "uint256", "name": "y", "type": "uint256"},
+            ],
+            "indexed": False,
+            "internalType": "struct DepositEventContract.T",
+            "name": "value",
+            "type": "tuple",
+        },
+    ],
+    "name": "Deposit",
+    "type": "event",
+}
+
+ABI_EVENT_DEPOSIT = {
+    "anonymous": False,
+    "inputs": [
+        {
+            "indexed": True,
+            "internalType": "address",
+            "name": "from",
+            "type": "address",
+        },
+        {
+            "indexed": True,
+            "internalType": "bytes32",
+            "name": "id",
+            "type": "bytes32",
+        },
+        {
+            "indexed": False,
+            "internalType": "uint256",
+            "name": "value",
+            "type": "uint256",
+        },
+    ],
+    "name": "Deposit",
+    "type": "event",
+}
+ABI_FUNCTION_DEPOSIT_STRUCT = {
+    "inputs": [{"internalType": "bytes32", "name": "id", "type": "bytes32"}],
+    "name": "depositStruct",
+    "outputs": [],
+    "stateMutability": "payable",
+    "type": "function",
+}
+ABI_FUNCTION_DEPOSIT_VALUE = {
+    "inputs": [{"internalType": "bytes32", "name": "id", "type": "bytes32"}],
+    "name": "depositValue",
+    "outputs": [],
+    "stateMutability": "payable",
+    "type": "function",
+}
+AMBIGUOUS_EVENT_WITH_TUPLE_CONTRACT_ABI = [
+    ABI_EVENT_DEPOSIT_WITH_TUPLE,
+    ABI_EVENT_DEPOSIT,
+    ABI_EVENT_TRANSFER,
+    ABI_FUNCTION_DEPOSIT_STRUCT,
+    ABI_FUNCTION_DEPOSIT_VALUE,
+]
+
+
+def test_get_abi_from_event(ambiguous_event_contract: "Contract") -> None:
+    expected_event_abi = {
+        "anonymous": False,
+        "inputs": [
+            {
+                "indexed": False,
+                "internalType": "uint256",
+                "name": "arg0",
+                "type": "uint256",
+            }
+        ],
+        "name": "LogSingleArg",
+        "type": "event",
+    }
+
+    event_get_abi_result = ambiguous_event_contract.events[
+        "LogSingleArg(uint256)"
+    ]._get_event_abi()
```

Comment:
This is a stylistic nitpick, but it seems a little smelly to me to test a private method. It looks like `_get_event_abi` is used to assign the abi attribute on `ContractEvent`. Would it be possible to do something like: 

```suggestion
event_get_abi_result = ambiguous_event_contract.events[
    "LogSingleArg(uint256)"
].abi
```

Feel free to take or leave if that doesn't make sense.

Reply:
This is definitely better as `.abi` 👍